### PR TITLE
Add npm-dependency-duplications and npm-dependency-vulnerabilities actions

### PR DESCRIPTION
**Pull request by Monteth**
This PR adds 2 new github actions: `npm-dependency-duplications` and `npm-dependency-vulnerabilities`.
They are both based on the [javascript-action](https://github.com/actions/javascript-action) template and use [github actions toolkit](https://github.com/actions/toolkit) to write action code in javascript.

### CODE REVIEW

File: actions/npm-dependency-duplications/action.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: 'Npm security audit'
+description: 'Checks if project dependencies have any known vulnerabilities.'
+inputs:
+  path:
+    default: .
+    description: NPM project localization
+    required: false
+runs:
+  using: 'node16'
```
Comment (radekmie) [inline actions/npm-dependency-duplications/action.yml]: Can this be parametrized on the callee side?

File: actions/npm-dependency-duplications/action.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+name: 'Npm security audit'
```
Comment (radekmie) [inline actions/npm-dependency-duplications/action.yml]: I'd rather use "npm" everywhere as that's the suggested name.

**Review by radekmie — CHANGES_REQUESTED** [review]

File: actions/npm-dependency-duplications/action.yml (hunk 1 above)
Comment (Monteth) [reply actions/npm-dependency-duplications/action.yml]: I am pretty sure we don't want to configure that on the callee side, as it's only for setting the execution environment for our `index.js` file [docs](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runsusing). You probably want to configure the npm version for `npm dedupe` (and `npm audit`), but we are calling npm commands in the existing environment.
You can configure which node version you want to use by running `actions/setup-node` with desired parameters before using our action. 

Example:
```yaml
    steps:
      - uses: actions/checkout@v2
      - name: Set Node.js 15
        uses: actions/setup-node@v2.5.1
        with:
          node-version: 15
      - uses: ./actions/npm-dependency-duplications
        with:
          path: ./actions/npm-dependency-duplications/test/positive
```

will run `npm dedupe` check with node@15.

Comment (radekmie) [reply actions/npm-dependency-duplications/action.yml]: That was exactly my point. Great!

**Review by radekmie — APPROVED** [review]

File: .github/workflows/check_dependency_duplications.yml [hunk 3]
```diff
@@ -1,39 +1,65 @@
...
+
 
 jobs:
-  check_duplications:
-    name: Check dependency duplications
+  units:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set Node.js 16
```
Comment (wadamek65) [inline .github/workflows/check_dependency_duplications.yml]: Same suggestion for other places too:
```suggestion
      - name: Set up Node.js 16
```

File: .github/workflows/check_dependency_duplications.yml [hunk 4]
```diff
@@ -1,39 +1,65 @@
-name: Check npm dependency duplications
-
+name: "npm-dependency-duplications test"
```
Comment (wadamek65) [inline .github/workflows/check_dependency_duplications.yml]: nitpick: use single quotes for consistency or remove them if they are not needed. Saw also occurrences in other yml files but it's very minor so fix at your own discretion.

File: .github/workflows/check_dependency_vulnerabilities.yml [hunk 5]
```diff
@@ -1,39 +1,64 @@
-name: Check npm dependency vulnerabilities
-
+name: "npm-dependency-vulnerabilities test"
```
Comment (wadamek65) [inline .github/workflows/check_dependency_vulnerabilities.yml:1]: Same as above.

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,4 @@
+
```
Comment (wadamek65) [inline .gitignore]: Was this on purpose? Also consider adding the vscode directories since `.idea` is here already.
```suggestion
```

**Review by wadamek65 — APPROVED** [review]

**Pull request merged**